multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/cf/v2/ApplicationCloudModelBuilder.java

```diff
@@ -202,12 +202,27 @@ protected Map<String, Map<String, Object>> getBindingParameters(Module module) {
     protected void addBindingParameters(Map<String, Map<String, Object>> result, RequiredDependency dependency, Module module) {
         Resource resource = getResource(dependency.getName());
         if (resource != null) {
-            MapUtil.addNonNull(result, resource.getName(), getBindingParameters(dependency, module.getName()));
+            Map<String, Object> bindingConfigParameters = getBindingConfigParameters(dependency, module.getName());
+            Map<String, Object> bindingNameParameters = getBindingNameParameters(dependency);
+
+            if (bindingConfigParameters == null && bindingNameParameters == null) {
+                MapUtil.addNonNull(result, resource.getName(), null);
+                return;
+            }
+
+            Map<String, Object> combinedParameters = new HashMap<>();
+            if (bindingConfigParameters != null) {
+                combinedParameters.putAll(bindingConfigParameters);
+            }
+            if (bindingNameParameters != null) {
+                combinedParameters.putAll(bindingNameParameters);
+            }
+            MapUtil.addNonNull(result, resource.getName(), combinedParameters);
```

Comment:
This way the binding name is mixed with the other binding parameters, then this binding name will be sent to the controller, some service brokers might not like it and start to fail.
Instead of nesting the maps, create a new class for Binding Name + Binding Parameters and use it here.